source/jquery.flot.js

```diff
@@ -2492,46 +2494,68 @@ Licensed under the MIT license.
             }
         }
 
-        // returns the data item the mouse is over/ the cursor is closest to, or null if none is found
-        function findNearbyItem(mouseX, mouseY, seriesFilter, radius, computeDistance) {
-            var i, j,
-                item = null,
-                smallestDistance = radius * radius + 1;
+        function findNearbyItems(mouseX, mouseY, seriesFilter, radius, computeDistance) {
+           var items = [];
 
-            for (var i = series.length - 1; i >= 0; --i) {
-                if (!seriesFilter(i)) continue;
+           findItem(mouseX, mouseY, seriesFilter, radius, computeDistance, items);
 
-                var s = series[i];
-                if (!s.datapoints) return;
+           executeHooks(hooks.clickHoverFindNearby, [mouseX, mouseY, seriesFilter, radius, computeDistance, items]);
```

Comment:
Finally getting back to looking at this PR again (sorry). So, I think the main thing I'd like to drive consensus on is the proper default behavior for this method (findNearbyItems), particularly when there are no discovered hooks. Intuitively it seems like this method should be able to discover multiple items (even when no hooks are discovered). I think this would mean that `findItem` would become `findItems`, and perhaps take a parameter that would cap the number of items to find. Then, `findNearbyItem` would no longer call `findNearbyItems` directly, but share a similar implementation to that of `findNearbyItems`. OR, we add another parameter to `findNearbyItems` that allows the user to cap how many items to return (this doesn't seem bad).

What do you think?

Reply:
I completely agree with that, I believe that if there's an interface that implies it'd return everything within "X" distance of a point, it should always do that and not just when there is a hook to provide more places to look. 

The issue I'm running into, is that this is moving beyond what I'm comfortable doing at this time. When I was initially writing this PR, I was trying to avoid reworking existing systems as much as possible. Of course, looking back it's obvious I was working to change a paradigm even when I simply thought I was extending a bit of functionality.

At this time I believe that it'd take me more time than I currently have to be happy with any re-engineering of the current `findItem` I would do.

Reply:
Totally understand. Sorry I've been such a roadblock.